### PR TITLE
Add e2e test case for downward API exposing pod UID

### PR DESCRIPTION
**Pull request by xiangpengzhao**
**What this PR does / why we need it**:
Pod UID is added to downward API env var in #48125 for 1.8. This PR adds a e2e test case for it.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
ref: #48125

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/common/downward_api.go [hunk 1]
```diff
@@ -187,6 +187,29 @@ var _ = framework.KubeDescribe("Downward API", func() {
...
+					FieldRef: &v1.ObjectFieldSelector{
+						APIVersion: "v1",
+						FieldPath:  "metadata.uid",
+					},
+				},
+			},
+		}
+
+		expectations := []string{
+			fmt.Sprintf("POD_UID=%v", podUID),
```
Comment (xiangpengzhao) [inline test/e2e/common/downward_api.go]: I should use regexp here. Will fix it tomorrow.

**Review by timothysc — APPROVED** [review]
/lgtm

**Pull request merged**